Dockerfile

```
@@ -1,34 +1,47 @@
-FROM node:12-stretch
-LABEL description="Debian image to build nativefier apps"
+FROM node:12-alpine
+LABEL description="Alpine image to build Nativefier apps"
 
-# Get wine32, not 64, to work around binary incompatibility with rcedit.
-# https://github.com/nativefier/nativefier/issues/375#issuecomment-304247033
-# Forced us to use Debian rather than Alpine, which doesn't do multiarch.
-RUN dpkg --add-architecture i386
 
 # Install dependencies
-RUN apt-get update \
-    && apt-get --yes install wine32 imagemagick \
-    && apt-get clean && rm -rf /var/lib/apt/lists/*
+RUN apk update \
+    && apk add bash wine imagemagick dos2unix \
+    && rm -rf /var/cache/apk/*
+
+WORKDIR /nativefier
 
 # Add sources
-COPY . /nativefier
+COPY . .
+
+# Fix line endings that may have gotten mangled in Windows
+RUN find ./icon-scripts ./src ./app -type f -print0 | xargs -0 dos2unix
 
 # Build nativefier and link globally
 WORKDIR /nativefier/app
 RUN npm install
 WORKDIR /nativefier
-RUN npm install && npm run build && npm link
+
+# Install (which will also install in ./app, and build, thanks to our `prepare` npm hook)
+# Also, running tests, to ensure we don't Docker build & publish broken stuff
+RUN npm install && npm run build && npm test && npm link
+
+# Cleanup test artifacts
+RUN rm -rf /tmp/nativefier*
 
 # Use 1000 as default user not root
 USER 1000
 
-# Run a {lin,mac,win} build: 1. to check installation was sucessful,
-# 2. to cache electron distributables and avoid downloads at runtime.
+# Run a {lin,mac,win} build
+# 1. to check installation was sucessful
+# 2. to cache electron distributables and avoid downloads at runtime
 RUN nativefier https://github.com/nativefier/nativefier /tmp/nativefier \
     && nativefier -p osx https://github.com/nativefier/nativefier /tmp/nativefier \
-    && nativefier -p windows https://github.com/nativefier/nativefier /tmp/nativefier \
-    && rm -rf /tmp/nativefier
+    && nativefier -p windows https://github.com/nativefier/nativefier /tmp/nativefier
+
+
+RUN echo Generated total cache size: $(du -sh ~/.cache/electron) \
+    && echo Generated total app size: $(du -sh /tmp/nativefier) \
+    && rm -rf /tmp/nativefier \
```

I'd remove this log, since we're removing `/tmp/nativefier` anyway, it's not of importance to end users.

```suggestion
    && rm -rf /tmp/nativefier \
```